src/analysis/processing/qgsrasteranalysisutils.cpp

```diff
@@ -479,6 +479,116 @@ double QgsRasterAnalysisUtils::varietyFromCellValues( std::vector<double> cellVa
   return uniqueValues.size();
 }
 
+double QgsRasterAnalysisUtils::nearestRankPercentile( std::vector<double> cellValues, int stackSize, double percentile )
```

Comment:
These signatures cause a copy of the cellValues vector -- I think you could potentially pass this as a non-const reference to avoid the copy